### PR TITLE
Upgrade to Swift 4

### PR DESCRIPTION
**Pull request by jgrandelli**

### CODE REVIEW

**Comment by jgrandelli** [comment]
Tests pass. Travis will probably fail because travis

File: Example/URBNJSONDecodable-Example.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -494,7 +505,8 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = "com.urbn.URBNJSONDecodable-Example";
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				SWIFT_VERSION = 3.0;
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (codecaffeine) [inline Example/URBNJSONDecodable-Example.xcodeproj/project.pbxproj]: I commented on the other ticket, but we might want to look into this at some point

**Review by codecaffeine — APPROVED** [review]

**Comment by jgrandelli** [comment]
updated tests passing @codecaffeine

File: Example/URBNJSONDecodable-Example.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -523,7 +536,8 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = "com.urbn.URBNJSONDecodable-ExampleTests";
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				SWIFT_VERSION = 3.0;
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (codecaffeine) [inline Example/URBNJSONDecodable-Example.xcodeproj/project.pbxproj]: `Default`

File: Example/URBNJSONDecodable-Example.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -539,7 +553,8 @@
...
```
Comment (codecaffeine) [inline Example/URBNJSONDecodable-Example.xcodeproj/project.pbxproj]: `Default`

**Review by codecaffeine — CHANGES_REQUESTED** [review]

**Review by codecaffeine — APPROVED** [review]

**Pull request merged**